File: src/main/java/com/dh/ondot/schedule/infra/exception/OdsayMissingParamException.java
package com.dh.ondot.schedule.infra.exception;

import com.dh.ondot.core.exception.BadRequestException;

import static com.dh.ondot.core.exception.ErrorCode.ODSAY_MISSING_PARAM;

public class OdsayMissingParamException extends BadRequestException {
    public OdsayMissingParamException(String detail) {
        super(ODSAY_MISSING_PARAM.getMessage().formatted(detail));
    }
    @Override
    public String getErrorCode() {
        return ODSAY_MISSING_PARAM.name();
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/infra/exception/OdsayMissingParamException.kt
package com.dh.ondot.schedule.infra.exception

import com.dh.ondot.core.exception.BadRequestException
import com.dh.ondot.core.exception.ErrorCode.ODSAY_MISSING_PARAM

class OdsayMissingParamException(detail: String) :
    BadRequestException(ODSAY_MISSING_PARAM.message.format(detail)) {
    override val errorCode: String get() = ODSAY_MISSING_PARAM.name
}
